src/ContractInteract/EIP20CoGateway.js

```diff
@@ -90,6 +94,128 @@ class EIP20CoGateway {
     this._redeemRawTx = this._redeemRawTx.bind(this);
   }
 
+  // TODO: docs
+  static setup(web3, config, txOptions = {}) {
+    EIP20CoGateway.validateSetupConfig(config);
+    validateConfigKeyExists(config, 'gateway', 'config');
+
+    const deployParams = Object.assign({}, txOptions);
+    deployParams.from = txOptions.from || config.deployer;
+
+    return EIP20CoGateway.deploy(
+      web3,
+      config.valueToken, // TODO: valueToken -> baseToken
+      config.utilityToken,
+      config.anchor,
+      config.bounty,
+      config.organization,
+      config.gateway,
+      config.burner,
+      config.messageBus,
+      config.gatewayLib,
+      deployParams,
+    );
+  }
+
+  // TODO: docs
+  static validateSetupConfig(config) {
+    validateConfigExists(config);
+    validateConfigKeyExists(config, 'deployer', 'config');
+    validateConfigKeyExists(config, 'bounty', 'config');
+    validateConfigKeyExists(config, 'organization', 'config');
+    validateConfigKeyExists(config, 'anchor', 'config');
+    validateConfigKeyExists(config, 'messageBus', 'config');
+    validateConfigKeyExists(config, 'gatewayLib', 'config');
+    validateConfigKeyExists(config, 'valueToken', 'config');
+    validateConfigKeyExists(config, 'utilityToken', 'config');
+    validateConfigKeyExists(config, 'burner', 'config');
+  }
+
+  // TODO: docs
+  static async deploy(
+    web3,
+    valueToken,
+    utilityToken,
+    anchor,
+    bounty,
+    membersManager,
+    gateway,
+    burner,
+    messageBus,
+    gatewayLib,
+    txOptions,
+  ) {
+    const tx = EIP20CoGateway.deployRawTx(
+      web3,
+      valueToken,
+      utilityToken,
+      anchor,
+      bounty,
+      membersManager,
+      gateway,
+      burner,
+      messageBus,
+      gatewayLib,
+    );
+
+    const _txOptions = txOptions;
+    if (!_txOptions.gas) {
+      _txOptions.gas = await tx.estimateGas();
+    }
+
+    return Utils.sendTransaction(tx, _txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new EIP20CoGateway(web3, address);
+    });
+  }
+
+  // TODO: docs
+  static deployRawTx(
+    web3,
+    valueToken,
+    utilityToken,
+    anchor,
+    bounty,
+    membersManager,
+    gateway,
+    burner,
+    messageBus,
+    gatewayLib,
+  ) {
+    const messageBusLibInfo = {
+      address: messageBus,
+      name: 'MessageBus',
+    };
+    const gatewayLibInfo = {
+      address: gatewayLib,
```

Comment:
Maybe name the arguments `messageBusAddress` and `gatewayLibAddress`? And possibly strip the `...Info` tail?

Reply:
Changed the argument names to the ones you proposed, but changed the name of `gatewayLibInfo` to `gatewayLibLinkInfo`, as I found to vague without any suffix.